tests/CMakeLists.txt

```diff
@@ -53,10 +53,14 @@ set(TEST_SOURCES
 	"${CMAKE_SOURCE_DIR}/rwgame/GameInput.cpp"
 	)
 
-ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
-
 add_executable(run_tests ${TEST_SOURCES})
 
+if(NOT LINK_STATIC)
+	target_compile_definitions(run_tests
+		PUBLIC
+			BOOST_TEST_DYN_LINK)
+endif()
+
```

Comment:
I feel like this should be a seperate PR.

I feel we should discuss what the standard cmake way of enforcing this is, I'd assume there are ways of doing this?